### PR TITLE
publish-docs

### PR DESCRIPTION
**Pull request by caleb-johnson**
Upload a docs artifact to the CI interface to make verifying and debugging docs more straightforward. This PR should prevent devs from having to build and verify docs locally.

You can see in [this PR](https://github.com/Qiskit-Extensions/quantum-serverless/actions/runs/5513506037?pr=756) that the docs for PRs are now available through the github GUI for direct download and inspection.

### CODE REVIEW

File: .github/workflows/docs-verify.yml [hunk 1]
```diff
@@ -20,3 +20,9 @@ jobs:
       - name: Build docs
         run: |
           tox -edocs
+      - name: Upload docs artifact
+        if: always()
```
Comment (caleb-johnson) [inline .github/workflows/docs-verify.yml:24]: This forces the build to upload even if it has errors. This is useful for debugging cryptic sphinx failures

**Review by Eric-Arellano — APPROVED** [review]
Looks great! But the PR title is misleading - this is not _publishing_ the docs, which usually means shipping code to _production_. It is only uploading the docs artifact in CI to GitHub

**Comment by IceKhan13** [comment]
already part of https://github.com/Qiskit-Extensions/quantum-serverless/pull/757

**Comment by caleb-johnson** [comment]
Covered by #757

**Pull request closed without merging**